File: api/DAO/ClienteDAO.cs
using api.DataBase;
using api.Helpers;
using api.Models;
using api.Repositories;
using MySql.Data.MySqlClient;

namespace api.DAO
{
    public class ClienteDAO : IClienteRepository
    {
        private readonly MySqlConnection _connection;

        public ClienteDAO()
        {
            _connection = Connection.GetInstance().GetConnection;
        }

        public void Cadastrar(Cliente cliente)
        {
            string saldoFormat = cliente.Saldo.ToString().Replace(",", ".");
            string query = $"insert into Cliente (nome, conta, senha, saldo) values ('{cliente.Nome}', {cliente.Conta}, '{cliente.Senha}', {saldoFormat})";
            SqlHelper sql = new(_connection);
            sql.Execute(query);
        }

        public Cliente? BuscaPorConta(long conta)
        {
            string query = $"select * from Cliente where conta={conta}";
            SqlHelper sql = new(_connection);

            var tb_clientes = sql.Query(query);
            var clientes = Cliente.DataTableConvert(tb_clientes);

            if (clientes.Count == 0) return null;

            return clientes[0];
        }

        public Cliente? BuscaPorId(int id)
        {
            string query = $"select * from Cliente where id={id}";
            SqlHelper sql = new(_connection);

            var tb_clientes = sql.Query(query);
            var clientes = Cliente.DataTableConvert(tb_clientes);

            if (clientes.Count == 0) return null;

            return clientes[0];
        }

        public Cliente? Login(LoginDados _params) 
        {
            string query = $"select * from Cliente where conta={_params.Conta} and senha='{_params.Senha}'";
            SqlHelper sql = new(_connection);

            var tb_clientes = sql.Query(query);
            var clientes = Cliente.DataTableConvert(tb_clientes);

            if (clientes.Count == 0) return null;

            return clientes[0];
        }
    }
}

File: api_node/src/dao/cliente-dao.ts
import { Connection } from 'mysql';
import { Conexao } from '../database/conexao';
import { DBService } from '../services/db-service';
import { Cliente, ClienteProps } from '../models/cliente-model';
import { IClinteRepository, LoginProps } from '../repositories/cliente-repository';

export class ClienteDAO implements IClinteRepository {
  private readonly _conexao: Connection;
  private readonly _dbService: DBService;

  constructor() {
    this._conexao = Conexao.getInstance().conexao();
    this._dbService = new DBService(this._conexao);
  }

  async login({ conta, senha }: LoginProps) {
    const consulta = `select * from Cliente where conta=${conta} and senha=${senha}`;
    const resposta = await this._dbService.execute(consulta) as ClienteProps[];
    
    if(!resposta.length) return null;
    
    return Cliente.convertMap(resposta)[0];
  }

  async buscarPorId(idCliente: number) {
    const consulta = `select * from Cliente where id=${idCliente}`;
    const resposta = await this._dbService.execute(consulta) as ClienteProps[];
    
    if(!resposta.length) return null;
    
    return Cliente.convertMap(resposta)[0];
  }

  async buscarPorConta(conta: number) {
    const consulta = `select * from Cliente where conta=${conta}`;
    const resposta = await this._dbService.execute(consulta) as ClienteProps[];
    
    if(!resposta.length) return null;
    
    return Cliente.convertMap(resposta)[0];
  }
}